tests\ui\native-library-link-flags\msvc-non-utf8-output.rs fails with lld-link as the linker

This test is either new or changed recently, it now fails when using -Clinker=lld-link

Example build failure on Chromium CI: https://logs.chromium.org/logs/chromium/buildbucket/cr-buildbucket/8781936910784713969/+/u/package_rust/stdout

```
  diff of stderr:
  
  -	error: linking with `link.exe` failed: exit code: 1181
  +	error: linking with `C:\b\s\w\ir\cache\builder\src\third_party\rust-toolchain-intermediate\llvm-host-install\bin/lld-link` failed: exit code: 1
  2	   |
  -	   = note: LINK : fatal error LNK1181: cannot open input file 'märchenhaft.obj'
  +	   = note: lld-link: error: could not open 'märchenhaft': no such file or directory
  4	           
  5	
  6	error: aborting due to previous error
```

The test has link.exe error messages hardcoded into it, which are wrong when not using link.exe.

We can disable this test when not using link.exe but it would be much better to make the error checking more flexible or explicitly define an output for ldd-link.

How would one go about the latter?

--- a/tests/ui/native-library-link-flags/msvc-non-utf8-output.rs
+++ b/tests/ui/native-library-link-flags/msvc-non-utf8-output.rs
@@ -1,6 +1,5 @@
 // build-fail
-// compile-flags:-C link-arg=märchenhaft
+// compile-flags:-C link-arg=⦺ⅈ⽯⭏⽽◃⡽⚞
 // only-msvc
-// error-pattern:= note: LINK : fatal error LNK1181:
-// normalize-stderr-test "(\s*\|\n)\s*= note: .*\n" -> "$1"
+// normalize-stderr-test "(?:.|\n)*(⦺ⅈ⽯⭏⽽◃⡽⚞)(?:.|\n)*" -> "$1"
 pub fn main() {}

--- a/tests/ui/native-library-link-flags/msvc-non-utf8-output.stderr
+++ b/tests/ui/native-library-link-flags/msvc-non-utf8-output.stderr
@@ -1,7 +1,1 @@
-error: linking with `link.exe` failed: exit code: 1181
-   |
-   = note: LINK : fatal error LNK1181: cannot open input file 'märchenhaft.obj'
-           
-
-error: aborting due to previous error
-
+⦺ⅈ⽯⭏⽽◃⡽⚞

